gef-ui/server/controllers/unissued-facilities/index.js

```diff
@@ -270,11 +299,82 @@ const postChangeUnissuedFacilityPreview = async (req, res) => {
   }
 };
 
+/**
+ * post function for changing changedToIssued facility back to unissued
+ * changes date fields to null and adds the months of cover back to database if changing facility back to unissued
+ * works in collaboration with facilities model in portal-api
+ * @param {req}
+ * @param {res}
+ * @returns {res}
+ */
+const postChangeIssuedToUnissuedFacility = async (req, res) => {
+  const {
+    body, params, query, session,
+  } = req;
+  const { dealId, facilityId } = params;
+  const { user } = session;
+  const { _id: editorId } = user;
+  let { facilityType } = query;
+  const hasBeenIssuedErrors = [];
+  facilityType = facilityType || FACILITY_TYPE.CASH;
+  const facilityTypeString = FACILITY_TYPE[facilityType?.toUpperCase()].toLowerCase();
+
+  // Don't validate form if user clicks on 'return to application` button
+  if (!body.hasBeenIssued) {
+    hasBeenIssuedErrors.push({
+      errRef: 'hasBeenIssued',
+      errMsg: `Select if your bank has already issued this ${facilityTypeString} facility`,
+    });
+
+    return res.render('partials/issued-facility-to-unissued.njk', {
+      facilityType: facilityTypeString,
+      errors: validationErrorHandler(hasBeenIssuedErrors),
+      dealId,
+    });
+  }
+
+  try {
+    if (body.hasBeenIssued === 'false') {
+      const { details } = await api.getFacility(facilityId);
+
```

Comment:
Please validate whether `details` object is not null.

Reply:
done